aliyun/src/main/java/org/apache/iceberg/aliyun/AliyunClientFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aliyun;
+
+import com.aliyun.oss.OSS;
+import java.io.Serializable;
+import java.util.Map;
+
+public interface AliyunClientFactory extends Serializable {
+
+  /**
+   * Create an aliyun OSS client.
+   *
+   * @return oss client.
+   */
+  OSS ossClient();
```

Comment:
Can you make this `newOSSClient`? We typically prefix factory method names with `new`.

Reply:
Rework and follow up this guide